src/celeritas/ext/detail/HitProcessor.cc

```diff
@@ -86,6 +86,36 @@ get_step_status(DetectorStepOutput const& out, size_type step_index)
 //---------------------------------------------------------------------------//
 }  // namespace
 
+//---------------------------------------------------------------------------//
+/*!
+ * Restore the G4Track from the reconstruction data. Takes ownership of the
+ * user information by unsetting it in the original track.
+ */
+HitProcessor::GeantTrackReconstructionData::GeantTrackReconstructionData(
+    G4Track const& track)
+    : track_id{track.GetTrackID()}
+    , user_info{track.GetUserInformation()}
+    , creator_process{track.GetCreatorProcess()}
+{
+    CELER_EXPECT(*this);
+    // Clear user information so that it doesn't get deleted with the G4Track
+    track.SetUserInformation(nullptr);
```

Comment:
Of course geant4 has `mutable G4VUserTrackInformation* fpUserInformation` 🙄 

Even though it means people can avoid const casts, we're actually changing the member data. I think constructing with `G4Track&` and calling `GeantTrackReconstructionData` using a `const_cast` would make it clearer in the code that we're actually changing the track.

Reply:
Should we also propagate the mutable ref all the way to `LocalTransporter::Push`?